src/Compilers/CSharp/Test/Semantic/FlowAnalysis/LocalFunctions.cs

```diff
@@ -7,6 +8,136 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     public class LocalFunctions : FlowTestBase
     {
         [Fact]
+        public void InvalidBranchOutOfLocalFunc()
+        {
+            var comp = CreateCompilationWithMscorlib(@"
+class C
+{
+    public static void M()
+    {
+        label1:
+
+        L();
+        L2();
+        L3();
+
+        void L()
+        {
+            goto label1;
+        }
+
+        void L2()
+        {
+            break;
+        }
+
+        void L3()
+        {
+            continue;
+        }
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (14,13): error CS0159: No such label 'label1' within the scope of the goto statement
+                //             goto label1;
+                Diagnostic(ErrorCode.ERR_LabelNotFound, "goto").WithArguments("label1").WithLocation(14, 13),
+                // (19,13): error CS0139: No enclosing loop out of which to break or continue
+                //             break;
+                Diagnostic(ErrorCode.ERR_NoBreakOrCont, "break;").WithLocation(19, 13),
+                // (24,13): error CS0139: No enclosing loop out of which to break or continue
+                //             continue;
+                Diagnostic(ErrorCode.ERR_NoBreakOrCont, "continue;").WithLocation(24, 13),
+                // (6,9): warning CS0164: This label has not been referenced
+                //         label1:
+                Diagnostic(ErrorCode.WRN_UnreferencedLabel, "label1").WithLocation(6, 9));
+        }
+
+        [Fact]
+        [WorkItem(13762, "https://github.com/dotnet/roslyn/issues/13762")]
+        public void AssignsInAsync()
+        {
+            var comp = CreateCompilationWithMscorlib46(@"
+using System.Threading.Tasks;
+
+class C
+{
+    public static void M2()
+    {
+        int x, y, z;
+        L1();
+        x++;
+        y++;
+        z++;
+
+        async void L1()
+        {
+            x = 0;
+            await Task.Delay(0);
+            y = 0;
+        }
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (10,9): error CS0165: Use of unassigned local variable 'x'
+                //         x++;
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(10, 9),
```

Comment:
>ERR_UseDefViolation [](start = 37, length = 19)

I think this error is incorrect. Control does not return to the caller until `x` has been assigned.